File: frontend/lib/data/models/depot.dart
import 'stock.dart';

class Depot {
  final int? id;
  final String libelleDepot;
  final String? code;
  final String? nom;
  final String? adresse;
  final String? ville;
  final String? codePostal;
  final String? telephone;
  final int? magasinId;
  final String? magasinNom;
  final double? latitude;
  final double? longitude;
  final int? capaciteStockage;
  final bool? actif;
  final List<Stock>? stocks;

  Depot({
    this.id,
    required this.libelleDepot,
    this.code,
    this.nom,
    this.adresse,
    this.ville,
    this.codePostal,
    this.telephone,
    this.magasinId,
    this.magasinNom,
    this.latitude,
    this.longitude,
    this.capaciteStockage,
    this.actif = true,
    this.stocks,
  });

  factory Depot.fromJson(Map<String, dynamic> json) {
    return Depot(
      id: json['id'],
      libelleDepot: json['libelleDepot'] ?? '',
      code: json['code'],
      nom: json['nom'],
      adresse: json['adresse'],
      ville: json['ville'],
      codePostal: json['codePostal'],
      telephone: json['telephone'],
      magasinId: json['magasinId'],
      magasinNom: json['magasinNom'],
      latitude: json['latitude']?.toDouble(),
      longitude: json['longitude']?.toDouble(),
      capaciteStockage: json['capaciteStockage'],
      actif: json['actif'],
      stocks: json['stocks'] != null
          ? (json['stocks'] as List).map((e) => Stock.fromJson(e)).toList()
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'id': id,
      'libelleDepot': libelleDepot,
      'code': code,
      'nom': nom,
      'adresse': adresse,
      'ville': ville,
      'codePostal': codePostal,
      'telephone': telephone,
      'magasinId': magasinId,
      'latitude': latitude,
      'longitude': longitude,
      'capaciteStockage': capaciteStockage,
      'actif': actif,
    };
  }
}

File: backend/src/main/java/com/example/backend/model/Depot.java
package com.example.backend.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "depots")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Depot {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(name = "libelle_depot", nullable = false)
    private String libelleDepot;
    
    @Column(nullable = false)
    private String code;
    
    @Column(nullable = false)
    private String nom;
    
    private String adresse;
    
    private String ville;
    
    @Column(name = "code_postal")
    private String codePostal;
    
    private String telephone;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "magasin_id", nullable = false)
    private Magasin magasin;
    
    // Coordonnées GPS pour le routing OSRM
    @Column(nullable = false)
    private Double latitude;
    
    @Column(nullable = false)
    private Double longitude;
    
    @Column(name = "capacite_stockage")
    private Integer capaciteStockage;
    
    private Boolean actif = true;
    
    @Column(nullable = false)
    private Integer version = 1;
    
    @Column(name = "created_at")
    private LocalDateTime createdAt;
    
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
    
    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }
    
    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}
